clog/readers.py

```diff
@@ -284,6 +287,8 @@ def __init__(self,
                 raise Exception("Last n lines can be only used with new kafka tailer")
             self._stream = stream + " " + str(lines)
             self._automagic_recovery = False
+        if protocol_opts:
+            self._stream += ''.join([' {0}={1}'.format(k, v) for k, v in protocol_opts.items()])
```

Comment:
I agree with yzernik. I would change also L288 to not modify _stream.
What about we keep lines and protocol_opts as attributes and we change L320 to serialize the proper connection message?
We could add a new method to generate the connection msg and add unit tests for it.